src/eval.ml

```diff
@@ -87,6 +87,16 @@ and find_rule : sym -> stack -> (term * stack) option = fun s stk ->
     in
     match_args r.lhs stk
   in
+  (* EXPE *)
```

Comment:
The construction of a tree should be done at the time of rule declaration, not at the time of evaluation.

Reply:
Fixed in bac2fc5